op-acceptance-tests/tests/isthmus/operator_fee/balance_reader.go

```diff
@@ -0,0 +1,89 @@
+package isthmus
+
+import (
+	"context"
+	"math/big"
+
+	"github.com/ethereum-optimism/optimism/devnet-sdk/testing/systest"
+	"github.com/ethereum-optimism/optimism/op-service/predeploys"
+	"github.com/ethereum/go-ethereum/common"
+	gethTypes "github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/stretchr/testify/require"
+)
+
+// BalanceReader provides methods to read balances from the chain
+type BalanceReader struct {
+	client *ethclient.Client
+	t      systest.T
+	logger log.Logger
+}
+
+// NewBalanceReader creates a new BalanceReader instance
+func NewBalanceReader(t systest.T, client *ethclient.Client, logger log.Logger) *BalanceReader {
+	return &BalanceReader{
+		client: client,
+		t:      t,
+		logger: logger,
+	}
+}
+
+// SampleBalances reads all the relevant balances at the given block number
+// and returns a BalanceSnapshot containing the results
+func (br *BalanceReader) SampleBalances(ctx context.Context, blockNumber *big.Int, walletAddr common.Address) *BalanceSnapshot {
+	br.logger.Debug("Sampling balances",
+		"block", blockNumber,
+		"wallet", walletAddr.Hex())
+	// Get the block to determine coinbase
+	var header *gethTypes.Header
+	var err error
+
+	if blockNumber == nil {
+		header, err = br.client.HeaderByNumber(ctx, nil)
+	} else {
+		header, err = br.client.HeaderByNumber(ctx, blockNumber)
+	}
```

Comment:
Couldn't this just be simplified?

```suggestion
	header, err = br.client.HeaderByNumber(ctx, blockNumber)
```